apps/studio/src/server/modules/user/user.service.ts

```diff
@@ -0,0 +1,13 @@
+import { db } from "../database"
+
+export const isUserDeleted = async (email: string) => {
+  const lowercaseEmail = email.toLowerCase()
+  const user = await db
+    .selectFrom("User")
+    .where("email", "=", lowercaseEmail)
+    .select(["deletedAt"])
+    // Email is a unique field in User table
+    .executeTakeFirst()
+
+  return !!user?.deletedAt
```

Comment:
```suggestion
  return user !== undefined && user.deleted !== null
```

Wondering if a more precise comparison might be better; since `deletedAt` is a `Timestamp` type which includes `string`, if `user.deletedAt === ''`, this would still pass the initial check even if it's not intended.

Original is still fine though, just a nit

Reply:
The deletedAt field which translated back can only be of type Date, null or undefined so wouldn't be an issue